src/features/staking/types/StakingTypes.ts

```diff
@@ -1,8 +1,6 @@
-export type StatusStaking =
-  | "unapproved"
-  | "approving"
-  | "approved"
-  | "readyToStake"
-  | "staking"
-  | "success"
-  | "failed";
+export enum StakeOrUnstake {
+  STAKE = "stake",
+  UNSTAKE = "unstake",
+}
+
+export type WagmiLoadingStatus = "error" | "idle" | "success" | "loading";
```

Comment:
If you wanted to be fancy:

```typescript
import { useQuery } from "wagmi";

type Status = ReturnType<typeof useQuery>["status"];
```

This way if the statuses ever change in the future you're okay (but this is extremely unlikely).

Reply:
I created a new issue for this to fix in a future PR https://github.com/airswap/airswap-voter-rewards/issues/47

Reply:
I indeed want to be fancy in this case